### PR TITLE
feat: add method to capture viewport

### PR DESCRIPTION
**Pull request by aluedeke**
adding the possibility to capture viewport screenshots for safari

### CODE REVIEW

File: lib/protocol/index.js [hunk 1]
```diff
@@ -148,7 +148,7 @@ const COMMANDS = {
   'Page.setShowPaintRects': ['result'],
   'Page.setEmulatedMedia': ['media'],
   'Page.snapshotNode': ['nodeId'],
-  'Page.snapshotRect': ['x', 'y', 'with', 'height', 'coordinateSystem'],
+  'Page.snapshotRect': ['x', 'y', 'width', 'height', 'coordinateSystem'],
```
Comment (mykola-mokhnach) [inline lib/protocol/index.js:151]: good catch

File: lib/remote-debugger.js [hunk 2]
```diff
@@ -307,6 +307,26 @@ class RemoteDebugger extends EventEmitter {
     });
   }
 
+  async captureScreenshot() {
+    log.debug('Capturing screenshot');
+    if (!this.rpcClient) {
+      throw new Error(`rpcClient is undefined. Has 'initRpcClient' been called before?`);
+    }
+
+    const width = await this.executeAtom('execute_script', ['return window.innerWidth', []]);
```
Comment (mykola-mokhnach) [inline lib/remote-debugger.js:316]: I believe we can return both in a single request

File: lib/remote-debugger.js [hunk 3]
```diff
@@ -307,6 +307,26 @@ class RemoteDebugger extends EventEmitter {
     });
   }
 
+  async captureScreenshot() {
```
Comment (mykola-mokhnach) [inline lib/remote-debugger.js:310]: shall we make it possible to provide the screenshotting rect as an argument?

File: lib/remote-debugger.js [hunk 4]
```diff
@@ -307,6 +307,26 @@ class RemoteDebugger extends EventEmitter {
...
+    log.debug('Capturing screenshot');
+    if (!this.rpcClient) {
+      throw new Error(`rpcClient is undefined. Has 'initRpcClient' been called before?`);
+    }
+
+    const width = await this.executeAtom('execute_script', ['return window.innerWidth', []]);
+    const height = await this.executeAtom('execute_script', ['return window.innerHeight', []]);
+
+    return await this.rpcClient.send('Page.snapshotRect', {
```
Comment (mykola-mokhnach) [inline lib/remote-debugger.js:319]: what does this method return?

File: lib/remote-debugger.js [hunk 5]
```diff
@@ -307,6 +307,26 @@ class RemoteDebugger extends EventEmitter {
...
+      appIdKey: this.appIdKey,
+      pageIdKey: this.pageIdKey,
+      x: 0,
+      y: 0,
+      width,
+      height,
+      coordinateSystem: 'Viewport'
```
Comment (mykola-mokhnach) [inline lib/remote-debugger.js:326]: what other coordinate systems are possible?

**Comment by mykola-mokhnach** [comment]
would it be possible to add an integration test for the new feature?

File: lib/remote-debugger.js [hunk 6]
```diff
@@ -307,6 +307,26 @@ class RemoteDebugger extends EventEmitter {
...
+    }
+
+    const width = await this.executeAtom('execute_script', ['return window.innerWidth', []]);
+    const height = await this.executeAtom('execute_script', ['return window.innerHeight', []]);
+
+    return await this.rpcClient.send('Page.snapshotRect', {
+      appIdKey: this.appIdKey,
+      pageIdKey: this.pageIdKey,
+      x: 0,
+      y: 0,
```
Comment (KazuCocoa) [inline lib/remote-debugger.js:323]: Does this mean the start point is `(0,0)`?

File: lib/remote-debugger.js [hunk 7]
```diff
@@ -7,6 +7,7 @@ import _ from 'lodash';
 import B from 'bluebird';
 import path from 'path';
 import AsyncLock from 'async-lock';
+import { type } from 'os';
```
Comment (mykola-mokhnach) [inline lib/remote-debugger.js:10]: this import is not used

**Comment by aluedeke** [comment]
@mykola-mokhnach all the 4 new tests are passing accroding too logs, no clue whats causing those problems :/

File: lib/remote-debugger.js (hunk 6 above)
Comment (aluedeke) [reply lib/remote-debugger.js]: full viewport and page screenshots do start at 0,0 but that behavior can be overridden in the latest version

**Review by mykola-mokhnach — APPROVED** [review]

**Pull request merged**